### PR TITLE
feat: match skills with free-text add

### PR DESCRIPTION
**Pull request by lessp**
**Ticket:** [#136](https://trello.com/c/GuVPUpe6/136-finish-match-skills)
**Intent:**

**How to test (optional):**

### All of the following commands should pass.

- [x] `npm test`
- [x] `npm run lint`
- [ ] `npm run cypress`

### Browser testing

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] IE11
- [ ] Edge

### CODE REVIEW

File: src/hooks/useGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }, {})
+
+    return items.map(item => ({
+      ...item,
+      xy: [item.xy[0] + rowSpace[item.row] / 2, item.xy[1]],
+    }))
+  }
+
+  let prevElement: GridItem | undefined
+  const gridItems = items.map((child: UseGridItem) => {
```
Comment (lundeeell) [inline src/hooks/useGrid.tsx]: Looks like a perfect case for a reduce

Comment (lundeeell) [inline src/hooks/useGrid.tsx]: Shouldn't be needed typ type child as items already is

**Comment by lessp** [comment]
> AddTraits stopped to work with this branch.
> 

Hmm, what doesn't work? Seems to work over here!

> Would be nice to focus input field when pressing "Lägg till ny kompetens +"-tag, now I have to tap twice in order to get my phone keyboard.
> 

True, will add!

> Remove typescript errors from code.
> 

Hmm, where are the Typescript-errors?

> When adding a new competence, I couldn't deselect it, it tried to add it twice (and gave me console error)

Good catch! Will investigate!

**Comment by lundeeell** [comment]
> > Remove typescript errors from code.
> 
> Hmm, where are the Typescript-errors?

In  src/components/TagListAnimations.tsx

File: src/components/TagListAnimations.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            font-size: 10px;
+            line-height: 10px;
+            border-radius: 10px;
+          }
+        `}
+      />
+      <div className="list" style={{ width: ROW_WIDTH, height: ROW_WIDTH }}>
+        {transitions.map<any>(({ item, props: { xy }, ...rest }: any) => {
+          return (
+            <a.div
```
[the remark this answers is not in the document]
Comment (lessp) [reply src/components/TagListAnimations.tsx]: Removed this for now, there's a local copy!

File: src/hooks/useGrid.tsx (hunk 1 above)
Comment (lessp) [reply src/hooks/useGrid.tsx]: See above

Comment (lessp) [reply src/hooks/useGrid.tsx]: See above

**Comment by lessp** [comment]
@lundeeell, please have another look now and make sure everything works!

**Pull request merged**